heartbeat/monitors/active/synthetic/script.go

```diff
@@ -0,0 +1,113 @@
+package synthetic
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/elastic/beats/v7/heartbeat/eventext"
+	"github.com/elastic/beats/v7/heartbeat/monitors"
+	"github.com/elastic/beats/v7/heartbeat/monitors/jobs"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/google/uuid"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func init() {
+	monitors.RegisterActive("script", create)
+	monitors.RegisterActive("synthetic/script", create)
+}
+
+type synthproxyReq struct {
+	ScriptParams map[string]interface{} `json:"script_params"`
+	Script string `json:"script"`
+}
+
+func create(name string, cfg *common.Config) (js []jobs.Job, endpoints int, err error) {
+	config := defaultConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, 0, err
+	}
+
+	job := monitors.MakeSimpleCont(func(event *beat.Event) error {
+		dataDir := filepath.Join("data",  uuid.New().String())
+		scriptPath := filepath.Join(dataDir, "script.js")
+		os.MkdirAll(dataDir, 0755)
+		//defer os.RemoveAll(dataDir)
+
+		err = ioutil.WriteFile(scriptPath, []byte(config.Script), 0644)
+		if err != nil {
+			return fmt.Errorf("could not write script file %s: %w", scriptPath, err)
+		}
+
+		logp.Info("Start script job")
+		cmd := exec.Command(
+			"node",
```

Comment:
What do you think about shipping a bin file from our package? and hearbeat can invoke our node script using shell? 

```
./elastic-synthetic  /path/to/script.js

```

Reply:
There is also other cool way to do this, Hearbeat can pipe this code and Node.js can read it from stdin. Advantage is multiple sources can be passed, we can also align on a separator that would provide distinction to journeys.

Reply:
++ to this, already done